v3_async_wip/v3_async_wip/exceptions.py

```diff
@@ -0,0 +1,28 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module defines exceptions used by other modules"""
```

Comment:
Very much work in progress - recoverable vs non-recoverable hasn't been implemented meaningfully (yet)